mixer/cmd/build.go

```diff
@@ -568,3 +619,39 @@ func init() {
 		append(externalDeps[buildUpdateCmd],
 			externalDeps[buildImageCmd]...)...)
 }
+
+func checkFormatMismatch(b *builder.Builder) error {
+	//Skip check if offline
+	if builder.Offline {
+		return nil
+	}
+
+	hostFormat, upstreamFormat, err := b.GetHostAndUpstreamFormats()
+	if err != nil {
+		return err
+	}
+
+	if hostFormat == "" {
+		fmt.Println("Warning: Unable to determine host format. Running natively may fail.")
+	} else if hostFormat != upstreamFormat {
+		fmt.Println("Warning: The host format and mix upstream format do not match.",
+			"Mixer may be incompatible with this format; running natively may fail.")
+	}
+
+	return nil
+}
+
+func checkFormatBump(b *builder.Builder) error {
```

Comment:
so, `checkFormatMismatch` prints warnings and returns `nil` but this function doesn't print anything and returns `err`s. Any reason for the inconsistency?

Reply:
Those code chunks where extracted from the older root preRun(). They have similar naming, but they don't perform the same function. The format check main purpose is to inform the user, it only fails on `GetHostAndUpstreamFormats()` because this is an early way to stop the build if there is no connection on offline was not used. We could ignore the error and continue.
On the other hand, the `checkFormatBump()` is an actual check. If the check fails, the execution should stop.
Maybe I could rename `checkFormatMismatch()` to `informFormatMismatch()` if that would make it less confusing.

Reply:
I think either names work, if you want more options `PrintFormatMismatch` works too IMO.